lib/rack_middleware/robots_generator.rb

```diff
@@ -5,13 +5,16 @@ class RobotsGenerator
   # Use the config/robots.txt in production.
   # Disallow everything for all other environments.
   def self.call(env)
+    return [404, {}, []] if env[:current_marketplace].nil?
```

Comment:
Yes, I think it makes sense. The env[:current_marketplace] shouldn't be nil in any sensible scenario.